akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala

```diff
@@ -411,6 +411,12 @@ object Source {
       combineRest(2, rest.iterator)
     })
 
+  def transpose[T, U](sources: Seq[Source[T, _]]): Source[Seq[T], NotUsed] = sources match {
```

Comment:
We should think about the name as well. While this certainly can be imagined as transposing a "matrix" with finite rows and infinite columns to a "matrix" with infinite rows and finite columns, I don't expect people finding/understanding this easily. I think more people will think about this as a kind of Zip but with homogeneous types.